src/daft-connect/src/spark_analyzer.rs

```diff
@@ -47,11 +45,15 @@ use crate::{
 #[derive(Clone)]
 pub struct SparkAnalyzer<'a> {
     pub session: &'a ConnectSession,
+    expr_resolver: ExprResolver,
 }
 
 impl SparkAnalyzer<'_> {
     pub fn new(session: &ConnectSession) -> SparkAnalyzer<'_> {
-        SparkAnalyzer { session }
+        SparkAnalyzer {
+            session,
+            expr_resolver: ExprResolver::new(),
```

Comment:
Are there plans on adding parameters to the constructor? If not, we don't need to store this inside of SparkAnalyzer, and could just call ExprResolver::resolve_expr directly.